src/etl/NFTHelpers.cpp

```diff
@@ -368,3 +372,30 @@ getNFTData(ripple::TxMeta const& txMeta, ripple::STTx const& sttx)
             return {{}, {}};
     }
 }
+
+std::vector<NFTsData>
+getNFTDataFromObj(
+    std::uint32_t const seq,
+    std::string const& key,
+    std::string const& blob)
+{
+    std::vector<NFTsData> nfts;
+    ripple::STLedgerEntry const sle = ripple::STLedgerEntry(
+        ripple::SerialIter{blob.data(), blob.size()},
+        ripple::uint256::fromVoid(key.data()));
+
+    if (sle.getFieldU16(ripple::sfLedgerEntryType) != ripple::ltNFTOKEN_PAGE)
+        return nfts;
+
+    auto const owner = ripple::AccountID::fromVoid(key.data());
+    for (ripple::STObject const& node : sle.getFieldArray(ripple::sfNFTokens))
+    {
+        std::optional<ripple::Blob> uri;
+        if (node.isFieldPresent(ripple::sfURI))
+            uri = node.getFieldVL(ripple::sfURI);
+        nfts.emplace_back(
+            node.getFieldH256(ripple::sfNFTokenID), seq, owner, uri);
```

Comment:
you can probably move `owner` and `uri` to be more efficient (@godexsoft double check me please). This function will be called a lot during the initial ledger download. A little speed up will go a long way here.